extensions/typescript/src/features/completionItemProvider.ts

```diff
@@ -18,15 +18,25 @@ class MyCompletionItem extends CompletionItem {
 	document: TextDocument;
 	position: Position;
 
-	constructor(entry: CompletionEntry) {
+	constructor(position: Position, document: TextDocument, entry: CompletionEntry) {
 		super(entry.name);
 		this.sortText = entry.sortText;
 		this.kind = MyCompletionItem.convertKind(entry.kind);
+		this.position = position;
+		this.document = document;
 		if (entry.replacementSpan) {
 			let span: protocol.TextSpan = entry.replacementSpan;
 			// The indexing for the range returned by the server uses 1-based indexing.
 			// We convert to 0-based indexing.
 			this.textEdit = TextEdit.replace(new Range(span.start.line - 1, span.start.offset - 1, span.end.line - 1, span.end.offset - 1), entry.name);
+		} else {
+			const text = document.getText(new Range(position.line, Math.max(0, position.character - entry.name.length), position.line, position.character));
```

Comment:
This assume that the replacement text will always start with the replaced text. @dbaeumer is this a safe assumption to make. It worked in all cases in my testing, but we may want to scope the fix by only applying it to certain kinds of completion entries